src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1123,40 +1132,96 @@ export class ResultsViewPageStore {
+                    return client.getAllSampleIdsInSampleListUsingGET({
+                        sampleListId: spec.sampleListId
+                    });
+                }));
+                for (let i = 0; i < results.length; i++) {
+                    ret[sampleListsToQuery[i].studyId] = ret[sampleListsToQuery[i].studyId] || {};
+                    const sampleMap = ret[sampleListsToQuery[i].studyId];
+                    results[i].map(sampleId => {
+                        sampleMap[sampleId] = true;
+                    });
+                }
+                return ret;
         }
-        return ret;
+
     }, {});
 
-    @computed get studyToSampleListId(): { [studyId: string]: string } {
-        return this.samplesSpecification.reduce((map, next) => {
-            if (next.sampleListId) {
-                map[next.studyId] = next.sampleListId;
-            }
-            return map;
-        }, {} as {[studyId: string]: string});
-    }
+    // @computed get studyToSampleListId(): { [studyId: string]: string } {
+    //     return this.samplesSpecification.reduce((map, next) => {
+    //         if (next.sampleListId) {
+    //             map[next.studyId] = next.sampleListId;
+    //         }
+    //         return map;
+    //     }, {} as {[studyId: string]: string});
+    // }
+
+    readonly studyToSampleListId = remoteData<{ [studyId: string]: string }>({
+        await:()=>[
+            this._sampleSpecification
+        ],
+        invoke: async ()=>{
+            return this._sampleSpecification.result!.reduce((map, next) => {
+                if (next.sampleListId) {
+                    map[next.studyId] = next.sampleListId;
+                }
+                return map;
+            }, {} as {[studyId: string]: string});
+        }
+    })
+
+
+    readonly _sampleSpecification = remoteData({
+       invoke: async ()=>{
+
+           // is this a sample list category query?
+           // if YES, we need to derive the sample lists by:
+           // 1. looking up all sample lists in selected studies
+           // 2. using those with matching category
+           if (!this.sampleListCategory) {
+               return this.samplesSpecification
+           } else {
+               const allSampleLists = await Promise.all(this.samplesSpecification.map((spec)=>{
+                   return client.getAllSampleListsInStudyUsingGET({
+                       studyId: spec.studyId,
+                       projection: 'SUMMARY'
+                   })
+               }));
+
+               const specs = allSampleLists.reduce((aggregator:SamplesSpecificationElement[], sampleLists)=>{
+                   //find the sample list matching the selected category using the map from shortname to full category name :(
+                   const matchingList = _.find(sampleLists,(list)=>list.category===SampleListCategoryMap[this.sampleListCategory!]);
```

Comment:
pull `SampleListCategoryMap[this.sampleListCategory!]` out of the loops since it doesnt change